### PR TITLE
CBAPI - 3869 - Enhance the watchlist-feeds-reports guide

### PR DESCRIPTION
**Pull request by dganev-cb**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been reviewed and added/updated if needed (for bug fixes/features)
- [x] Code follows the style guidelines of this project (PEP8, clean code, linter).
- [x] A self-review of the code has been done.

## What is the ticket or issue number?
CBAPI-3869

## Pull Request Description
I wanted to create some quick snippets that you can just take and run since I found myself digging and creating those all over again, so I've put them into the guide. I can see that some of the code is being duplicated and I am not sure how to structure it but whenever I look for something IOC-related I just scroll down to the IOC section of that document (I believe the other people are doing the same thing). I am open to suggestions on how to structure the whole thing.

### CODE REVIEW

File: docs/watchlists-feeds-reports.rst [hunk 1]
```diff
@@ -191,6 +191,94 @@ Both IOCs and reports may include a ``link`` property, which is used by the Carb
...
+.. code-block:: python
+
+    >> from cbc_sdk.enterprise_edr import Watchlist, Report, IOC_V2
+    >> ioc_id = "<ioc_id>"
+    >> report_id = "<report_id>"
+    >> watchlist = cbcsdk.select(Watchlist, "<watchlist_id>")
+    <cbc_sdk.enterprise_edr.threat_intelligence.Watchlist: id <watchlist_id>> @ https://<CBCInstanceURL>
+    >> ioc = IOC_V2.create_regex(cbcsdk, None, "process_name", r"(^/usr/.*$)|(^/bin/.*$)")
+    >> reports = watchlist.reports
+    >> report = [report_ for report_ in reports if report_.id == report_id][0]
```
Comment (avanbrunt-cb) [inline docs/watchlists-feeds-reports.rst:276]: Hmm. This seems odd that we don't have a way to fetch the report directly for a watchlist as the API is available without the watchlist_id. I'd expect something like `cb.select(Report, "id", from_watchlist=True)`

Comment (avanbrunt-cb) [reply docs/watchlists-feeds-reports.rst:276]: Any thoughts on creating a story to implement this behavior?

**Review by avanbrunt-cb — APPROVED** [review]

File: docs/watchlists-feeds-reports.rst (hunk 1 above)
Comment (dganev-cb) [reply docs/watchlists-feeds-reports.rst:276]: Yes, I also think it's odd that this is not being supported however, you need to provide the Watchlist ID as well as the Report ID to select it so more like the current behavior with the `feed_id`.
```python
cb.select(Report, "id", watchlist_id="...")
```

Comment (avanbrunt-cb) [reply docs/watchlists-feeds-reports.rst:276]: Why do you need the Watchlist id? or is that because of the CBC SDK side?

Comment (dganev-cb) [reply docs/watchlists-feeds-reports.rst:276]: If we don't have an `watchlist_id` or `feed_id` how do we know where to look for that report?

Comment (dganev-cb) [reply docs/watchlists-feeds-reports.rst:276]: Actually just saw the `from_watchlist` which is set to a boolean when the real value should be a string. I will have to check the logic more thoughtfully to create a proper solution for this.

**Review by emitreva-cb — APPROVED** [review]

**Comment by dganev-cb** [comment]
Not merging until the release of the new version, also I have to add the new parts about the Report in https://github.com/carbonblack/carbon-black-cloud-sdk-python/pull/292

**Pull request merged**